### ISSUE TITLE
Example Included exits immediately

### ISSUE THREAD

**Issue by Excigma** [jaxcore/bumblebee-hotword-node#3]
I may have missed out something, but when I do `node index`, the immediately exits and does not log anything

(There has been activity from 20 days ago, so I doubt other people are able to reproduce)

![image](https://user-images.githubusercontent.com/30280397/81392884-6f223200-9173-11ea-8b47-a43aca2b3e9e.png)

I am using:
```js
// This is in index.js
const bumblebee = new BumbleBee();
bumblebee.addHotword('bumblebee');

bumblebee.on('hotword', function (hotword) {
	console.log('Hotword Detected:', hotword);
});

bumblebee.start();
```
from the https://github.com/jaxcore/bumblebee-hotword-node#quick-start

NodeJS Version: v13.14.0

I have sox installed and added to path:
![image](https://user-images.githubusercontent.com/30280397/81392904-78ab9a00-9173-11ea-9058-1d1ecbc2c946.png)

Another question, is there a way to stream audio data into this module, or do I _have_ to use my device microphone?

**Comment by dsteinman** [comment]
If you run your script with

`NODE_ENV=dev node index`

You'll get more debug output.

I've actually never run bumblebee in Windows before.  I do have another unreleased project in the works where I'm attempting to run bumblebee and DeepSpeech in an electron app, so that will remove the need for sox or any nodejs microphone libraries.

> Another question, is there a way to stream audio data into this module, or do I have to use my device microphone?

Yeah, the last pull request added this functionaality.

```
bumblebee.processAudio(inputFrame, inputSampleRate);
```

inputFrame is a Float 32 array from a microphone stream -- eg. the microphone stream from Chrome or some other source.  If you can find a solution that doesn't use sox I'd be happy to replace it.

**Comment by Excigma** [comment]
My main use case actually isn't inputting audio using the mic, it's from a stream, I will try to pipe the output to this then report back.

I will still need to convert the stream to the correct type that this module accepts and I am still trying to find a way to do it.

Thanks for your fast response though

**Comment by dsteinman** [comment]
You'll want to use the pcm-convert library to convert the stream to float32

https://www.npmjs.com/package/pcm-convert


You can see how I used it to convert the microphone stream in [bumblebee-node.js](https://github.com/jaxcore/bumblebee-hotword-node/blob/master/lib/bumblebee-node.js) Line 142:

```
// records as int16, convert back to float for porcupine
let float32arr = pcmConvert(data, 'int16 mono le', 'float32');
this.processAudio(float32arr, sampleRate);
this.emit('data', data, sampleRate);
```

**Comment by Excigma** [comment]
Thanks, using processAudio works, however I must call bumblebee.start() for it to work, This causes the microphone to be recording, can I start it without recording?

Edit: Also is it possible to add hotwords from say https://github.com/Picovoice/porcupine/tree/master/resources/keyword_files

**Comment by dsteinman** [comment]
Oh I forgot, when you call `.start()` you can supply an audio stream object, which will in turn run the float conversion code above.

```bumblebee.start(stream, sampleRate);```

So instead of using processAudio directly, try calling just start() with your audio stream object and the bitrate.

Only the wasm files can be used, unfortunately not many of the free ones created by Picovoice are suitable as an Alexa style hotword:

https://github.com/Picovoice/porcupine/tree/master/resources/keyword_files/wasm

I practically begged them to open source a "computer" hotword but they weren't very receptive.

**Comment by Excigma** [comment]
Thank you, it works, closing this issue
Even though there's an amount of limited words, there are still more options which is always good.

Edit: To convert the ppn file, I think just open it in a hex reader and copy it over, I'll report back if it doesn't work, thank you for the help

**Comment by dsteinman** [comment]
To convert the ppn file you use the xxd command line program to generate the byte array:

```xxd -i -g 1 file.ppn output.txt```

Then use the data in output.txt to create a new .js file like the ones here:

https://github.com/jaxcore/bumblebee-hotword-node/tree/master/hotwords

```
module.exports = new Uint8Array([
  ... byte array ...
]);
```

Then to apply the hotword to bumblebee use:

```
let newhotworddata = require('pathtojsfile');
bumblebee.addHotword("new_hotword", newhotworddata);
```

**Comment by dsteinman** [comment]
FYI it looks like they've updated Porcupine since I last updated the ppn files.  if the file format changed, then the files at https://github.com/Picovoice/porcupine/tree/master/resources/keyword_files/wasm might not work as-is.  You'd have to use git to revert the files to some time before this commit https://github.com/Picovoice/porcupine/commit/e66ca5aad955cf89974238a735c0b54c077c4d4e

**Comment by Excigma** [comment]
I've made a extremely patched node script to convert all the ppn files to js files, unfortunately it seems like some words are overly-sensitive and trigger on all sounds.

From reading your code it seems like you had the same problem with "terminator", and this probably can't be resolved. 

Thank you very much for all the help you've given me, it is much appreciated.

**Comment by SteTR** [comment]
Hello, I'm having the same issue where the example code exits immediately. 
**Versions:**
OS: Ubuntu 18.04,5
Bumblebee-hotword-node: 0.0.10
SoX: 14.4.2

Everything is the same as Excigma's original post.
I did try the basic example in the README and that was functioning properly.

**Comment by dsteinman** [comment]
Can you try using the latest version by cloning this repo?

I made a fix [here](https://github.com/jaxcore/bumblebee-hotword-node/commit/02b9c00f70a2f6c3652407cb51f659a964f91e2e) that corrects the location of `sox`, but I forgot to publish this as a new release.

**Comment by SteTR** [comment]
I just tried the example again with the cloned repository and it's functioning properly. The problem occurs when I'm trying to use the package via npm install and on a separate project.

**Comment by dsteinman** [comment]
Okay, I've just published master to npm as version 0.1.0  (previous version was 0.0.10).  This should resolve it.

**Comment by SteTR** [comment]
Retried the example code in a separate project and it worked. Thank you!
